cli/cmd/org/rename.go

```diff
@@ -26,65 +25,66 @@ func RenameCmd(ch *cmdutil.Helper) *cobra.Command {
 				return err
 			}
 
-			fmt.Println("Warn: Renaming an org would invalidate dashboard URLs")
-
-			if !cmd.Flags().Changed("org") && ch.Interactive {
-				orgNames, err := OrgNames(ctx, ch)
-				if err != nil {
-					return err
-				}
-
-				name, err = cmdutil.SelectPrompt("Select org to rename", orgNames, "")
-				if err != nil {
-					return err
+			resp, err := client.GetOrganization(ctx, &adminv1.GetOrganizationRequest{Name: name})
+			if err != nil {
+				if st, ok := status.FromError(err); ok && st.Code() == codes.NotFound {
+					ch.PrintfError("org %q doesn't exist, run 'rill org list' to see available orgs", name)
+					return nil
 				}
+				return err
 			}
+			org := resp.Organization
 
-			if ch.Interactive {
-				err = cmdutil.SetFlagsByInputPrompts(*cmd, "new-name")
-				if err != nil {
-					return err
-				}
+			// Build update request
+			req := &adminv1.UpdateOrganizationRequest{Name: org.Name}
+
+			if cmd.Flags().Changed("new-name") {
+				flagSet = true
+				req.NewName = &newName
 			}
 
-			if newName == "" {
-				return fmt.Errorf("please provide valid org new-name, provided: %q", newName)
+			if cmd.Flags().Changed("display-name") {
+				flagSet = true
+				req.DisplayName = &displayName
 			}
 
-			if !force {
-				msg := fmt.Sprintf("Do you want to rename org \"%s\" to \"%s\"?", color.YellowString(name), color.YellowString(newName)) // nolint:gocritic // Because it uses colors
-				ok, err := cmdutil.ConfirmPrompt(msg, "", false)
-				if err != nil {
-					return err
-				}
-				if !ok {
-					return nil
-				}
+			if !flagSet {
+				ch.PrintfError("no changes requested please specify --new-name or --display-name\n")
+				return nil
 			}
 
-			updatedOrg, err := client.UpdateOrganization(ctx, &adminv1.UpdateOrganizationRequest{
-				Name:    name,
-				NewName: &newName,
-			})
+			// Update org
+			updatedOrg, err := client.UpdateOrganization(ctx, req)
 			if err != nil {
 				return err
 			}
 
-			err = dotrill.SetDefaultOrg(newName)
-			if err != nil {
-				return err
+			// Print results
+			ch.PrintfSuccess("Updated organization\n")
+			if req.NewName != nil {
+				ch.Printf("Updated name: %s to %s\n", name, updatedOrg.Organization.Name)
+			}
+			if req.DisplayName != nil {
+				ch.Printf("Updated display name: %s to %s\n", org.DisplayName, updatedOrg.Organization.DisplayName)
```

Comment:
Previously you had a check where it printed a warning if it already has a display name, but the user only updated the unique name, i.e. something like `req.NewName != nil && req.DisplayName == nil && org.DisplayName != ""`.

Not a cause to prompt or error, but it's a nice warning because often someone would want to change both `Name` and `DisplayName` in that case.